src/user-guide/writing-workflows/runtime.rst

```diff
@@ -453,3 +453,222 @@ results in exactly the same task behaviour, via inheritance from root,
+:ref:`EventHandling`.
+
+If a task with configured retries is *killed* (by ``cylc kill``
+it goes to the *held* state so that the operator can decide
+whether to release it and continue the retry sequence or to abort the retry
+sequence by manually resetting it to the *failed* state.
+
+
+.. _EventHandling:
+
+Event Handling
+--------------
+
+* Task events (e.g. task succeeded/failed) are configured by
+  :cylc:conf:`task events <[runtime][<namespace>][events]>`.
+* Workflow events (e.g. workflow started/stopped) are configured by
+  :cylc:conf:`workflow events <[scheduler][events]>`
+
+.. cylc-scope:: flow.cylc[runtime][<namespace>]
+
+Cylc can call nominated event handlers - to do whatever you like - when certain
+workflow or task events occur. This facilitates centralized alerting and automated
+handling of critical events. Event handlers can be used to send a message, call
+a pager, or whatever; they can even intervene in the operation of their own
+workflow using cylc commands.
+
+To send an email, use the built-in setting :cylc:conf:`[events]mail events`
+to specify a list of events for which notifications should be sent. (The
+name of a registered task output can also be used as an event name in
+this case.) E.g. to send an email on (submission) failed and retry:
+
+.. code-block:: cylc
+
+   [runtime]
+       [[foo]]
+           script = """
+               test ${CYLC_TASK_TRY_NUMBER} -eq 3
+               cylc message -- "${CYLC_WORKFLOW_NAME}" "${CYLC_TASK_JOB}" 'oopsy daisy'
+           """
+           execution retry delays = PT0S, PT30S
+           [[[events]]]
+               mail events = submission failed, submission retry, failed, retry, oops
+           [[[outputs]]]
+               oops = oopsy daisy
+
+By default, the emails will be sent to the current user with:
+
+- ``to:`` set as ``$USER``
+- ``from:`` set as ``notifications@$(hostname)``
+- SMTP server at ``localhost:25``
+
+These can be configured using the settings:
+
+.. cylc-scope:: flow.cylc[runtime][<namespace>]
+
+- :cylc:conf:`[mail]to` (list of email addresses)
+- :cylc:conf:`[mail]from`
+
+.. cylc-scope::
+
+By default, a cylc workflow will send you no more than one task event email every
+5 minutes - this is to prevent your inbox from being flooded by emails should a
+large group of tasks all fail at similar time. This is configured by
+:cylc:conf:`[scheduler][mail]task event batch interval`.
+
+Event handlers can be located in the workflow ``bin/`` directory;
+otherwise it is up to you to ensure their location is in ``$PATH`` (in
+the shell in which the :term:`scheduler` runs). They should require little
+resource and return quickly - see :ref:`Managing External Command Execution`.
+
+.. cylc-scope:: flow.cylc[runtime][<namespace>]
+
+Task event handlers can be specified using the
+``[events]<event> handler`` settings, where
+``<event>`` is one of:
+
+.. TODO - Add link to replaced link of states.
```

Comment:
Did you want to do this TODO in this PR?

Reply:
Relates to #163

Reply:
Yes...But
```suggestion
 .. TODO - Add link to replaced list of states. 
```